supervisor/api/network.py

```diff
@@ -209,20 +209,17 @@ async def interface_update(self, request: web.Request) -> None:
         for key, config in body.items():
             if key == ATTR_IPV4:
                 interface.ipv4setting = replace(
-                    interface.ipv4setting
-                    or IpSetting(InterfaceMethod.STATIC, [], None, []),
+                    IpSetting(InterfaceMethod.STATIC, [], None, []),
                     **config,
                 )
```

Comment:
```suggestion
                interface.ipv4setting = IpSetting(**config)
```
There's no reason to do the `replace`as we aren't patch updating an existing object anymore, we're making a new object and using it. We should update the schema to includle `InterfaceMethod.STATIC, [], None, []` as the default values of their respective fields instead so we get those when the client doesn't provide their own.

Alternatively we can add a `from_dict` class method to those dataclasses and use that here, passing it `config`.

Reply:
We use also different standards in other places:

https://github.com/home-assistant/supervisor/blob/05e0c7c3ab949ec26c98f5dc98c4d8cf52ee3c0c/supervisor/host/configuration.py#L116

I'd prefer to keep this explicitly for the API.

Reply:
Ah I see you meant to add it to the schema validation. I've implemented that just now. It required vol.Maybe(), but in my mind this is a correct use-case for null values.